### DISCUSSION TITLE
Vue Component

### DISCUSSION THREAD

**Discussion by rxXxc** [hugerte/hugerte#31] [Q&A] [answered]
Hi,

I want to use in VueComponent where earlier I was having below code to use tinymce editor. What is the best way to achieve the same editor in VueComponent as I used in tinymce earlier.

import Editor from '@tinymce/tinymce-vue'

Also the package version is "@tinymce/tinymce-vue": "^5.1.0",

Please suggest the replacement in these case.

thanks in advance!

**Comment by carlosmintfan** [comment]
Hello,
Thank you for wanting to use HugeRTE. Unfortunately, I‘ve not yet forked the Vue integration to TinyMCE and adapted it to use HugeRTE. I’m very busy with other stuff at the moment, but I’ll get back to HugeRTE in about three weeks. I’d recommend you to use TinyMCE 6 until then. HugeRTE does not have any real improvements opposed to TinyMCE yet anyway. When I have more time, I’ll fork the Vue plugin as soon as possible. Thank you for your patience and understanding.

**Comment by rxXxc** [comment]
Hi @carlosmintfan , Thanks for the quick reply to my query and nice fork version of tinymce.  
Thanks!

**Comment by grndis** [comment]
Do you have any updates on this? I also need to use it to replace tinymce-vue.

**Reply by carlosmintfan** [reply]
Hello, today or at latest next week I‘m starting to have more time, then I‘ll get this done as soon as possible.

**Comment by carlosmintfan** [comment]
I‘ve done the basics today (replacing most TinyMCE refs by HugeRTE, removing apiKey and licenseKey props, switching from Tiny Cloud CDN to jsDelivr etc.) but the test stuff is a bit more complicated: I‘ll have to "rebrand" the https://github.com/hugerte/miniature also, publish it…but I hope that the Vue component should be ready next Monday or Tuesday. Svelte and other components will follow.

**Reply by grndis** [reply]
Thanks a lot for your amazing work on Hugerte!  I really appreciate all the effort you've put into this project. I'd be happy to help with testing when it's ready. Thank you again.

**Comment by carlosmintfan** [comment]
I can't promise yet when this will be ready. I'll give a little overview on the current status and the problem.
✅ HugeRTE is used instead of TinyMCE in most cases and variable names have been changed.
❌ Oops, I still have to migrate Utils.ts, but that shouldn't be a problem
✅ The demo works (I can't promise there are no bugs, should compare to the TinyMCE version)
✅ The storybook works
✅ Loader.ts and TestHelper.ts in src/test/ts/alien have been migrated from Sugar and Katamari to native methods (https://github.com/hugerte/hugerte/issues/35)
❌ src/test/ts/browser/InitTest.ts depends on `@tinymce/miniature`
❌ There is no published `@hugerte/miniature` at npm
❌ There are problems when replacing `@ephox` and `@tinymce` by `@hugerte` in the monorepo. https://github.com/hugerte/hugerte/issues/8 – now I've used jscodeshift instead and manually replaced stuff in package.json files and tsconfig – but there are still errors from TypeScript about types not being able to be resolved although...I updated the paths as far as I know. This will need more debugging...I don't know how much.
So, it's not much I'd have to do to be able to publish this – if I ignore the tests, which...is not good.

**Reply by carlosmintfan** [reply]
If anyone knows all TypeScript and yarn monorepo stuff well and feels like debugging it, you can look at https://github.com/hugerte/hugerte/tree/debugging-required-issue-8.
But I think I'll suggest a workaround: yarn linking.

**Reply by carlosmintfan** [reply]
Okay, so only VersionLoader.pLoadVersion is used from miniature in the InitTest, so I'll simply copy some code over...although I don't like the Apache license because of the requirement to make the files carry prominent notices about them having been modified if they were. (In a minified file, I may simply want to add a link to a different file containing all the legal notices, but Apache wants the files themselves to carry the notice.) But this is only testing stuff, thankfully.

**Reply by carlosmintfan** [reply]
I'm almost done with everything. Publish should happen tomorrow.

**Reply by carlosmintfan** [reply, marked as answer]
Done. https://npmjs.com/package/@hugerte/hugerte-vue. Please test thoroughly before using it.
